storm-server/src/main/java/org/apache/storm/container/oci/RuncLibContainerManager.java

```diff
@@ -0,0 +1,631 @@
+        if (imageName == null) {
+            LOG.error("Image name for {} is not configured properly; will not continue to launch the worker", topologyId);
+            return;
+        }
+
+        //set container ID to port + worker ID
+        String containerId = getContainerId(workerId, port);
+
+        //get manifest
+        ImageManifest manifest = imageTagToManifestPlugin.getManifestFromImageTag(imageName);
+        LOG.debug("workerId {}: Got manifest: {}", workerId, manifest.toString());
+
+        //get layers metadata
+        OciResource configResource = manifestToResourcesPlugin.getConfigResource(manifest);
+        LOG.info("workerId {}: Got config metadata: {}", workerId, configResource.toString());
+
+        saveRuncYaml(topologyId, port, containerId, imageName, configResource);
+
+        List<OciResource> layersResource = manifestToResourcesPlugin.getLayerResources(manifest);
+        LOG.info("workerId {}: Got layers metadata: {}", workerId, layersResource.toString());
+
+        //localize resource
+        String configLocalPath = ociResourcesLocalizer.localize(configResource);
+
+        List<String> ociEnv = new ArrayList<>();
+        List<String> args = new ArrayList<>();
+
+        ArrayList<OciLayer> layers = new ArrayList<>();
+
+        File file = new File(configLocalPath);
+        //extract env
+        List<String> imageEnv = extractImageEnv(file);
+        if (imageEnv != null && !imageEnv.isEmpty()) {
+            ociEnv.addAll(imageEnv);
+        }
+        for (Map.Entry<String, String> entry : env.entrySet()) {
+            ociEnv.add(entry.getKey() + "=" + entry.getValue());
+        }
+        LOG.debug("workerId {}: ociEnv: {}", workerId, ociEnv);
+
+        //extract entrypoint
+        List<String> entrypoint = extractImageEntrypoint(file);
+        if (entrypoint != null && !entrypoint.isEmpty()) {
+            args.addAll(entrypoint);
+        }
+        LOG.debug("workerId {}: args: {}", workerId, args);
+
+        //localize layers
+        List<String> layersLocalPath = ociResourcesLocalizer.localize((layersResource));
+        //compose layers
+        for (String layerLocalPath : layersLocalPath) {
+            OciLayer layer = new OciLayer(SQUASHFS_MEDIA_TYPE, layerLocalPath);
+            layers.add(layer);
+        }
+        LOG.debug("workerId {}: layers: {}", workerId, layers);
+        ArrayList<OciMount> mounts = new ArrayList<>();
+        setContainerMounts(mounts, topologyId, workerId, port);
+        LOG.debug("workerId {}: mounts: {}", workerId, mounts);
+
+        //calculate the cpusQuotas based on CPU_CFS_PERIOD and assigned CPU
+        Long cpusQuotas = null;
+        if (workerToCpu.containsKey(workerId)) {
+            cpusQuotas = workerToCpu.get(workerId) * CPU_CFS_PERIOD_US / 100;
+        }
+
+        Long memoryInBytes = null;
+        if (workerToMemoryMb.containsKey(workerId)) {
+            memoryInBytes = workerToMemoryMb.get(workerId) * 1024 * 1024L;
```

Comment:
Could this multiplication on the right side cause an overflow - int * int * long  when the first two integers are being multiplied.

Reply:
nice catch.